lib/Default/MysqlProperties.class.php

```diff
@@ -0,0 +1,71 @@
+<?php declare(strict_types=1);
+
+
+use Dotenv\Dotenv;
+
+
+class MysqlProperties {
+	private const CONFIG_MYSQL_HOST = "MYSQL_HOST";
+	private const CONFIG_MYSQL_USER = "MYSQL_USER";
+	private const CONFIG_MYSQL_PASSWORD = "MYSQL_PASSWORD";
+	private const CONFIG_MYSQL_PORT = "MYSQL_PORT";
+	private const CONFIG_MYSQL_DATABASE = "MYSQL_DATABASE";
+	private string $host;
+	private string $user;
+	private string $password;
+	private int $port;
+	private string $databaseName;
+
+	public function __construct() {
+		$configArray = !isset($_ENV["MYSQL_CONFIG_FROM_ENVIRONMENT"])
+			? Dotenv::createArrayBacked(ROOT)->load()
+			: $_ENV;
+		$this->parseConfigArray($configArray);
+	}
+
+	private function parseConfigArray(array $array) {
+		[
+			MysqlProperties::CONFIG_MYSQL_HOST => $this->host,
+			MysqlProperties::CONFIG_MYSQL_USER => $this->user,
+			MysqlProperties::CONFIG_MYSQL_PASSWORD => $this->password,
+			MysqlProperties::CONFIG_MYSQL_PORT => $port,
+			MysqlProperties::CONFIG_MYSQL_DATABASE => $this->databaseName,
+		] = $array;
+		$this->port = (int)$port;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getHost(): string {
```

Comment:
I'm a big proponent of making structs immutable whenever possible. Unfortunately PHP doesn't have the `final` keyword for fields, so making them public would make them mutable. It's also easier to mock these out by using methods, and it's generally considered best practice to make things immutable when they can be.

Reply:
Good point. For a moment I was confusing PHP with c++, where you can have const class members but still modify them in the constructor. Since that doesn't actually exist in PHP, private attributes with getters is :+1: with me!